autoprof/fit/base.py

```diff
@@ -66,13 +67,13 @@ def __init__(self, model, initial_state = None, relative_tolerance = 1e-3, **kwa
         self.loss_history = []
         self.message = ""
 
-    def fit(self):
+    def fit(self) -> Union[None,'BaseOptimizer']:
```

Comment:
That is good, I think it will always return a `BaseOptimizer` instance though, so the `None` option is not needed

Reply:
so I'll just remove the None?